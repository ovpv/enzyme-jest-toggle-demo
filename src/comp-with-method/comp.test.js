import React from 'react';

import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import MyComp from './comp';

configure({ adapter: new Adapter() });

describe('Component with save method', () => {
  it('should render properly', () => {
    const tree = renderer.create(<MyComp />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Method should not save if name passed is empty white space', () => {
    const wrapper = mount(<MyComp />);
    const instance = wrapper.instance();
    const name = '  ';
    const val = instance._onClickSaveLayout(name);
    expect(val).toBe(false);
  });
});
