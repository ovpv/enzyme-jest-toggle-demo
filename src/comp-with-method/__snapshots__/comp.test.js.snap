// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component with save method should render properly 1`] = `
<div>
  <input
    type="text"
  />
  <button
    onClick={[Function]}
  >
    Save
  </button>
</div>
`;
